### ISSUE TITLE
"Authorization is required to perform that action." during Google Workspace add-on review 

### ISSUE THREAD

**Issue by tventures02** [enuchi/React-Google-Apps-Script#130]
Hi, I've made a small Google Sheets add-on that I'm trying to get onto the Google Workplace Marketplace and am currently going through the review process.

The issue the testers are running into is when they try to run a function that calls the Session.getActiveUser().getEmail(), the add-on throws an error and in the logs, shows the "Authorization is required to perform that action" notice:

![image](https://user-images.githubusercontent.com/12650970/173616176-8e01034e-b8ac-41a1-9187-9ae551587628.png)

I have all the proper Oauth scopes in my appsscript.json file and when I deploy the add-on locally to a Google Sheet, everything works as intended. It's just when the tester reviews the app, they are having issues.

I cannot reproduce the reviewer's issue so I'm currently relegated to making a fix, republishing the add-on for review, and then seeing if that fix resolves the issue. I was hopefully someone here may have some insight on what the issue is.

Also, I've already published a Google Sheets add-on using this template as the boilerplate code about a year ago and did not encounter this issue. Any help would be greatly appreciated. Thank you.

**Comment by enuchi** [comment]
Not sure I can help much but the docs on `getActiveUser()` do explain some restrictions, which sound like it could be describing your situation. Are you calling this function during `onOpen`? 

https://developers.google.com/apps-script/reference/base/session#getActiveUser()

> Gets information about the current user. If security policies do not allow access to the user's identity, [User.getEmail()](https://developers.google.com/apps-script/reference/base/user#getEmail()) returns a blank string. The circumstances in which the email address is available vary: for example, the user's email address is not available in any context that allows a script to run without that user's authorization, like a simple onOpen(e) or onEdit(e) trigger, a custom function in Google Sheets, or a web app deployed to "execute as me" (that is, authorized by the developer instead of the user). However, these restrictions generally do not apply if the developer runs the script themselves or belongs to the same Google Workspace domain as the user.

**Comment by tventures02** [comment]
Thanks for the reply. I am not running getActiveUser from onOpen. It's running from a modal that the user has opened and then subsequently clicked a button.

I've also added the "https://www.googleapis.com/auth/userinfo.email" oauth scope to be appsscript.json file.

**Comment by enuchi** [comment]
Hmm, sounds annoying. Wouldn't surprise me if there were other reasons it didn't work that aren't well documented.

Not sure what you're doing with the email, but why don't you just put the call to `getActiveUser()` in a try block and move on in if it errors out. Or return an error to the modal and allow the user to input an email...

**Comment by tventures02** [comment]
I could do that but like I said before, I've used this repo in an add-on that is already published in the marketplace and it's using the same permissions/oauth scopes and working without an issue. 

BTW I really appreciate the work you've done here. It's really cool.

I was wondering if there's anything with the gas-client that has changed since 0.1.0 or something along those lines that might've affected something. I'm really reaching in the dark here.

**Comment by tventures02** [comment]
Yep, we've definitely seen those auth issues related to the multiple Google accounts with our users. I'll try simulating that but if it is that, it seems like that would be solved on the reviewers end. Anyhow, I'll try to simulate that to see if I can reproduce it. Thx

**Comment by tventures02** [comment]
I was able to reproduce the issue as suggested, video recorded the work around and sent it to the GSW reviewer. After some deliberation, they are able to approve the add-on. Thanks!
